internal/fileutils/fileutils.go

```diff
@@ -65,3 +67,32 @@ func CopyFile(srcPath, destPath string) error {
 
 	return dst.Sync()
 }
+
+// SymlinkResolutionError is the error returned when symlink resolution fails.
+type SymlinkResolutionError struct {
+	msg string
+	err error
+}
+
+func (e SymlinkResolutionError) Error() string {
+	return fmt.Sprintf("%s: %v", e.msg, e.err)
+}
+
+func (e SymlinkResolutionError) Unwrap() error {
+	return e.err
+}
```

Comment:
Do we need this is if using instead:

```go
func (e SymlinkResolutionError) Error() string {
	return fmt.Sprintf("%s: %w", e.msg, e.err)
}
```

Reply:
`fmt.Sprintf` doesn't have a `%w` specifier, only `fmt.Errorf` does